graphene_django/fields.py

```diff
@@ -62,14 +68,15 @@ def merge_querysets(cls, default_queryset, queryset):
         return queryset & default_queryset
 
     @classmethod
-    def resolve_connection(cls, connection, default_manager, args, iterable):
+    def resolve_connection(cls, connection, default_manager, args, info, iterable):
         if iterable is None:
             iterable = default_manager
         iterable = maybe_queryset(iterable)
         if isinstance(iterable, QuerySet):
             if iterable is not default_manager:
                 default_queryset = maybe_queryset(default_manager)
                 iterable = cls.merge_querysets(default_queryset, iterable)
+            iterable = optimize_queryset(iterable, info)
```

Comment:
Interestingly, this optimisation _does_ work because it's come after the call to `merge_querysets`!